receiver/prometheusreceiver/internal/metrics_adjuster.go

```diff
@@ -0,0 +1,169 @@
+	if len(current) != len(initial) {
+		// this shouldn't happpen, both lengths should always be 1.
+		ma.logger.Info("len(current points) != len(initial points)", len(current), len(initial))
+		return true
+	}
+	for i := 0; i < len(current); i++ {
+		if !ma.adjustPoint(metricType, current[i], initial[i]) {
+			return false
+		}
+	}
+	return true
+}
+
+// Types of metrics returned supported by prometheus:
+// - MetricDescriptor_CUMULATIVE_DOUBLE
+// - MetricDescriptor_GAUGE_DOUBLE
+// - MetricDescriptor_CUMULATIVE_DISTRIBUTION
+// - MetricDescriptor_GAUGE_DISTRIBUTION
+// - MetricDescriptor_SUMMARY
+func (ma *MetricsAdjuster) adjustPoint(metricType metricspb.MetricDescriptor_Type, current, initial *metricspb.Point) bool {
+	switch metricType {
+	case metricspb.MetricDescriptor_CUMULATIVE_DISTRIBUTION:
+		// note: sum of squared deviation not currently supported
+		initialDist := initial.GetDistributionValue()
+		currentDist := current.GetDistributionValue()
+		if currentDist.Count < initialDist.Count {
+			return false
+		}
+		currentDist.Count -= initialDist.Count
+		currentDist.Sum -= initialDist.Sum
+		ma.adjustBuckets(currentDist.Buckets, initialDist.Buckets)
+	case metricspb.MetricDescriptor_GAUGE_DISTRIBUTION:
+		// note: sum of squared deviation not currently supported
+		// note: we don't adjust bucket counts for gauge distributions
+		initialDist := initial.GetDistributionValue()
+		currentDist := current.GetDistributionValue()
+		if currentDist.Count < initialDist.Count {
+			return false
+		}
+		currentDist.Count -= initialDist.Count
+		currentDist.Sum -= initialDist.Sum
+		ma.adjustBuckets(currentDist.Buckets, initialDist.Buckets)
+	case metricspb.MetricDescriptor_SUMMARY:
+		// note: for summary, we don't adjust the snapshot
+		initialSummary := initial.GetSummaryValue()
+		currentSummary := current.GetSummaryValue()
+		initialCount := initialSummary.Count.GetValue()
+		currentCount := currentSummary.Count.GetValue()
+		if currentCount < initialCount {
+			return false
+		}
+		initialSum := initialSummary.Sum.GetValue()
+		currentSum := currentSummary.Sum.GetValue()
+		currentSummary.Count = &wrappers.Int64Value{Value: currentCount - initialCount}
+		currentSummary.Sum = &wrappers.DoubleValue{Value: currentSum - initialSum}
+	case metricspb.MetricDescriptor_CUMULATIVE_DOUBLE:
+		currentValue := current.GetDoubleValue()
+		initialValue := initial.GetDoubleValue()
+		if currentValue < initialValue {
+			// reset detected
+			return false
+		}
+		current.Value = &metricspb.Point_DoubleValue{DoubleValue: currentValue - initialValue}
+	default:
+		// this shouldn't happen
+		ma.logger.Info("adjust unexpect point type ", metricType.String(), ", skipping ...")
+	}
+	return true
+}
+
+func (ma *MetricsAdjuster) adjustBuckets(current, initial []*metricspb.DistributionValue_Bucket) bool {
+	if len(current) != len(initial) {
+		// this shouldn't happen
+		ma.logger.Info("len(current buckets) != len(initial buckets)", len(current), len(initial))
```

Comment:
should you change this  to Error?

Reply:
I generally avoid using Error unless something major is going on but that's based on Google prod convention - what is the convention for the oc agent when Error is appropriate?